### ISSUE TITLE
Only build on PRs and don't build on commits to stable or master

### ISSUE THREAD

**Issue by wilzbach** [dlang/ci#138]
- we don't display the status of Jenkins anywhere
- no one would notice that master broke - except it appears at a PR
- the PR already gets merged with its target branch
- it's a waste of resources
- we already do the same for AppVeyor and SemaphoreCI at dlang/dmd

**Comment by MartinNowak** [comment]
Not really IMO, remember that we don't rebuild pending PRs when master changes, because it's `O(N² / 2)`. To still catch bugs that are caused by semantic differences (not textural ones which would cause merge conflicts), the integration branch needs to be retested.
This could indeed be done less often, e.g. daily, because the chances are low and a bit of bisecting add much complexity.

**Comment by MartinNowak** [comment]
Same goes for AppVeyor and SemaphoreCI, let's please find a common policy for all of those.

**Comment by wilzbach** [comment]
It doesn't help anyone if no one looks at the errors which is the case at the moment.

**Comment by MartinNowak** [comment]
Yes, those would need a badge or sth. to be more visible, but someone will notice it eventually already.

**Comment by MartinNowak** [comment]
Also E-Mail notifications would be appropriate for a broken master.

**Comment by wilzbach** [comment]
I was the only one subscribed to these mails, but I can't fix everything. For CircleCi, everyone can click on subscribe on their platform to  receive such mails, but still I was the only one caring and fixing this when there was a failure a couple of days ago at Phobos.
With AppVeyor there were also a couple of random failures in the beginning.

> Yes, those would need a badge or sth. to be more visible, but someone will notice it eventually already.

Already there. Also these two CIs aren't mission critical and the free resources they offer are simply not enough to rebuild master on every PR. Daily builds are enabled though.

**Comment by wilzbach** [comment]
Also with so much frequency on GitHub and every PR merging with master a failure would immediately be discovered, but I haven't seen one case where merging to master made a difference or lead to a failure. The failures are caused by other things.

**Comment by MartinNowak** [comment]
Yes, daily build would still help to catch the rare issues that mostly happen with major refactorings. Since Kenji no longer works on the compiler, those have indeed become extremely rare.

**Comment by wilzbach** [comment]
On Buildkite we could in theory ignore building master ("Branch limiting") - https://buildkite.com/dlang/dmd/settings/repository
